### PR TITLE
Proposal to add rank_dropout scale

### PR DESCRIPTION
**Pull request by kovalexal**
Hi @KohakuBlueleaf! Thank you very much for your awesome library!

I was reading through your adapters code and was wondering - why there is no scaling when doing `rank_dropout`?

According to torch docs [nn.Dropout](https://pytorch.org/docs/stable/generated/torch.nn.Dropout.html) performs scaling of output values using the provided dropout probability to retain the overall normalization the same.

Screenshot 2023-09-28 at 13 58 39

I am not sure whether your implementation does have this scale intentionally or not (probably the further group normalization neutralizes the lack of this scale), but I've made several experiments, and adding this scaling may help training a bit faster if dropout is enabled.

### CODE REVIEW

**Comment by KohakuBlueleaf** [comment]
I just copied the dropout mechanism from kohya
I think what you pointed out is a good question

I will dig into this

**Comment by KohakuBlueleaf** [comment]
@kovalexal 2 things

1, Plz open a new PR to dev branch
2, Can you make this as an optional setting? so we will not change the default behavior (If you want, you can add some warning for this)

**Comment by kovalexal** [comment]
@KohakuBlueleaf sure, I'll add this as an optional parameter in the following commits to this PR.

**Comment by kovalexal** [comment]
@KohakuBlueleaf I added `rank_dropout_scale` parameter as an optional argument to all adapters.

I've also tried to run kohya_ss with this updated version, but from what I can tell, there is no ability to pass `rank_dropout` and `module_dropout` in kohya_ss (at least in their gui) according to this [script](https://github.com/bmaltais/kohya_ss/blob/31b26d59d3988980e4ecab464ae19efca06b80dd/lora_gui.py#L782).

In GUI there is an option "additional parameters" in the advanced tab, but when I write something like `--network_args "rank_dropout=0.3"` it overwrites the remaining  `--network_args` parameters in this [line](https://github.com/bmaltais/kohya_ss/blob/31b26d59d3988980e4ecab464ae19efca06b80dd/lora_gui.py#L791), so by default it switches training adapter to LoRA.

So there is a chance that we need to create PR to kohya_ss to be able to pass these parameters to your code.

I've also run several experiments with kohya_ss gui (`lokr-sd15` preset) and `rank_dropout` passed as 0.3 and from what I can tell, the resulting adapter does not produce meaningful results after training without the proposed scaling (see the following results). Is there a chance that we should enable this scale by default?

Scaling disabled (lokr-sd15, weight: 1.0):
LoKr_rank_dropout_0 3_noscale

LoKr_rank_dropout_0 3_noscale


Scaling enabled (lokr-sd15, weight: 1.0, rank_dropout: 0.3):
LoKr_rank_dropout_0 3_scale

LoKr_rank_dropout_0 3_scale

**Comment by laksjdjf** [comment]
For reference, sd-script already scales by default.
https://github.com/kohya-ss/sd-scripts/blob/2d87bb648f30adab00ceb38a0da786cd548d5ce7/networks/lora.py#L109
